datafusion/core/src/execution/context/mod.rs

```diff
@@ -1246,7 +1295,51 @@ impl QueryPlanner for DefaultQueryPlanner {
             .await
     }
 }
+/// Crates and registers a function from [CreateFunction] statement
+///
+/// It is intended to handle `CREATE FUNCTION` statements
+/// and interact with [SessionState] to registers new udfs.
+///
+/// Datafusion `SQL` dialect does not support `CREATE FUNCTION`
+/// in generic dialect, so dialect should be changed to `PostgreSQL`
```

Comment:
I agree supporting create function would be good in the generic dialect. I made this change in https://github.com/sqlparser-rs/sqlparser-rs/pull/1159 (and I plan a sqlparser release in the next few days)

Reply:
looking at `FunctionRegistry.remove` method, in order to implement it `FunctionRegistry` should be changed. Maybe that change is a bit too much for this PR. So i propose to remove `remove` from `FunctionFactory`  and add it back when we implement `unregister_ud*` methods in `FunctionRegistry`.

wdyt?

Reply:
I think we added the unregister_* functions recently (sorry our reviews crossed)